icu4c/source/i18n/formatted_string_builder.cpp

```diff
@@ -316,9 +343,9 @@ int32_t FormattedStringBuilder::prepareForInsertHelper(int32_t index, int32_t co
         fFields.heap.ptr = newFields;
         fFields.heap.capacity = newCapacity;
         fZero = newZero;
-        fLength += count;
+        fLength = newLength;
```

Comment:
Same two lines in this branch and in the "else" branch. I suggest you combine and move them to after the if-else.

Reply:
done